src/sound_check_menu.c

```diff
@@ -0,0 +1,2198 @@
+            gUnknown_020387B4[CRY_TEST_PANPOT] = 4;
+    }
+    if(gMain.newAndRepeatedKeys & 0x10) // _080BB192
+    {
+        gUnknown_020387B4[CRY_TEST_UNK0]++;
+        if(gUnknown_020387B4[CRY_TEST_UNK0] > 0xF7)
+            gUnknown_020387B4[CRY_TEST_UNK0] = 0;
+    }
+    else if(gMain.newAndRepeatedKeys & 0x20) // _080BB1B0
+    {
+        gUnknown_020387B4[CRY_TEST_UNK0]--;
+        if(gUnknown_020387B4[CRY_TEST_UNK0] < 0)
+            gUnknown_020387B4[CRY_TEST_UNK0] = 0xF7;
+    }
+}
+
+void sub_80BB1D4(void)
+{
+    u8 lrStr[5];
+    u8 rlStr[5];
+
+    memcpy(lrStr, gOtherText_LR, sizeof lrStr);
+    memcpy(rlStr, gOtherText_RL, sizeof rlStr);
+
+    sub_80BAE78(gUnknown_020387B4[CRY_TEST_UNK0], 7, 2, 3);
+
+    switch(gUnknown_083D03F8[gUnknown_020387B4[CRY_TEST_PANPOT]])
+    {
+        case 0x7F:
+            MenuPrint(lrStr, 7, 4);
+            break;
+        case -0x80:
+            MenuPrint(rlStr, 7, 4);
+            break;
+        default:
+            sub_80BAE78(gUnknown_083D03F8[gUnknown_020387B4[CRY_TEST_PANPOT]], 7, 4, 3);
+            break;
+    }
+    sub_80BAE78(IsSEPlaying(), 12, 8, 1);
+}
+
+void sub_80BB25C(u8 taskId)
+{
+    struct CryRelatedStruct cryStruct, cryStruct2;
+    u8 zero;
+
+    SetUpWindowConfig(&gWindowConfig_81E6C3C);
+    InitMenuWindow(&gWindowConfig_81E6CE4);
+    gUnknown_03005D34 = 1;
+    ResetSpriteData();
+    FreeAllSpritePalettes();
+
+    cryStruct.unk0 = 0x2000;
+    cryStruct.unk2 = 29;
+    cryStruct.paletteNo = 12;
+    cryStruct.yPos = 30;
+    cryStruct.xPos = 4;
+
+    zero = 0; // wtf?
+    gUnknown_03005E98 = 0;
+
+    while(sub_8119E3C(&cryStruct, 3) == FALSE);
+
+    cryStruct2.unk0 = 0;
+    cryStruct2.unk2 = 15;
+    cryStruct2.paletteNo = 13;
+    cryStruct2.xPos = 12;
+    cryStruct2.yPos = 12;
+
+    zero = 0; // wtf?
+    gUnknown_03005E98 = 0;
+
+    while(ShowPokedexCryScreen(&cryStruct2, 2) == FALSE);
+
+    MenuDrawTextWindow(0, 16, 5, 19);
+    sub_80BB494();
+    BeginNormalPaletteFade(0xFFFFFFFF, 0, 16, 0, 0);
+    REG_BG2HOFS = 0;
+    REG_BG2VOFS = 0;
+    REG_BG2CNT = 0xF01;
+    REG_BG3CNT = 0x1D03;
+    REG_DISPCNT = 0x1d40;
+    m4aMPlayFadeOutTemporarily(&gMPlay_BGM, 2);
+    TASK.FUNC = sub_80BB3B4;
+}
+
+void sub_80BB3B4(u8 taskId)
+{
+    sub_8119F88(3);
+
+    if(gMain.newKeys & 0x1)
+    {
+        sub_811A050(gUnknown_03005D34);
+    }
+    if(gMain.newKeys & 0x100)
+    {
+        StopCryAndClearCrySongs();
+    }
+    if(gMain.newAndRepeatedKeys & 0x40)
+    {
+        if(--gUnknown_03005D34 == 0)
+            gUnknown_03005D34 = 384; // total species
+        sub_80BB494();
+    }
+    if(gMain.newAndRepeatedKeys & 0x80)
+    {
+        if(++gUnknown_03005D34 > 384)
+            gUnknown_03005D34 = 1;
+        sub_80BB494();
+    }
+    if(gMain.newKeys & 0x2)
+    {
+        REG_DISPCNT = 0x7140;
+        REG_WIN0H = 0x11DF;
+        REG_WIN0V = 0x11F;
+        MenuZeroFillWindowRect(0, 0, 0x1D, 0x13);
```

Comment:
Personal opinion: numbers should be written as decimal, unless they're flags or "round" hex numbers like `0x2000`